controllers/util/solr_util.go

```diff
@@ -452,8 +452,15 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 	}
 
 	if solrCloud.Spec.SolrImage.ImagePullSecret != "" {
-		stateful.Spec.Template.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
-			{Name: solrCloud.Spec.SolrImage.ImagePullSecret},
+		if nil != customPodOptions {
```

Comment:
I'm not familiar with this code, but is it worth adding a check to make sure that .ImagePullSecrets exists in customPodOptions? (ie. is it possible to have a customPodOptions object without the ImagePullSecrets?)

Reply:
No need, if you append elements to a `nil` slice in Go, then it will create the slice too:
https://play.golang.org/p/lmkQFoIgT6G